### ISSUE TITLE
Context Menu

### ISSUE THREAD

**Issue by dickermoshe** [nank1ro/flutter-shadcn-ui#132]
### Steps to reproduce

1. Create Context Menu on Web
2. Right Click

### Expected results

Just See ShadCn Context Menu

### Actual results

See native browser one too

### shadcn_ui version

0.9.0

### Platform

Web

### Code sample

<details open><summary>Code sample</summary>

```dart
// https://mariuti.com/shadcn-ui/components/context-menu/
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/e326210f-b4cb-4cde-a0e9-6a9e71d38ad1

</details>

### Logs

<details open><summary>Logs</summary>

```console
N/A```

</details>


### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
N/A```

</details>

**Comment by nank1ro** [comment]
Which browser are you using?
To me on Arc and Chrome doesn't happen

**Comment by dickermoshe** [comment]
Chrome on Windows: 128.0.6613.114

**Comment by renanboni** [comment]
Since there is already an open issue with the Context Menu, I'm also facing a problem.

I'm using **ShadContextMenu** in a **ShadTable** (I want to have a edit/delete context menu when clicking on the ellipsis):
```

class EditVendorContextMenu extends StatefulWidget {
  const EditVendorContextMenu({super.key});

  @override
  State<EditVendorContextMenu> createState() => _EditVendorContextMenuState();
}

class _EditVendorContextMenuState extends State<EditVendorContextMenu> {
  final _isVisible = signal(false);

  @override
  Widget build(BuildContext context) {
    return ShadContextMenu(
      visible: _isVisible.watch(context),
      children: [
        ShadContextMenuItem(
          leading: const ShadImage.square(
            LucideIcons.pencil,
            size: 16,
          ),
          child: const Text("Edit"),
          onPressed: () {},
        ),
        const Divider(height: 8),
        ShadContextMenuItem(
          leading: const ShadImage.square(
            LucideIcons.trash,
            size: 16,
          ),
          child: const Text("Delete"),
          onPressed: () {},
        ),
      ],
      child: ShadButton.ghost(
        icon: const Icon(
          LucideIcons.ellipsis,
          size: 16,
        ),
        onPressed: () {
          _isVisible.value = !_isVisible.value;
        },
      ),
    );
  }

  @override
  void dispose() {
    _isVisible.dispose();
    super.dispose();
  }
}

```

However, when the context menu is dismissed by clicking outside, **_isVisible** will still be true, so I need to click on it twice to make it appear again.

**Comment by nank1ro** [comment]
> Since there is already an open issue with the Context Menu, I'm also facing a problem.
> 
> 
> 
> I'm using **ShadContextMenu** in a **ShadTable** (I want to have a edit/delete context menu when clicking on the ellipsis):
> 
> ```
> 
> 
> 
> class EditVendorContextMenu extends StatefulWidget {
> 
>   const EditVendorContextMenu({super.key});
> 
> 
> 
>   @override
> 
>   State<EditVendorContextMenu> createState() => _EditVendorContextMenuState();
> 
> }
> 
> 
> 
> class _EditVendorContextMenuState extends State<EditVendorContextMenu> {
> 
>   final _isVisible = signal(false);
> 
> 
> 
>   @override
> 
>   Widget build(BuildContext context) {
> 
>     return ShadContextMenu(
> 
>       visible: _isVisible.watch(context),
> 
>       children: [
> 
>         ShadContextMenuItem(
> 
>           leading: const ShadImage.square(
> 
>             LucideIcons.pencil,
> 
>             size: 16,
> 
>           ),
> 
>           child: const Text("Edit"),
> 
>           onPressed: () {},
> 
>         ),
> 
>         const Divider(height: 8),
> 
>         ShadContextMenuItem(
> 
>           leading: const ShadImage.square(
> 
>             LucideIcons.trash,
> 
>             size: 16,
> 
>           ),
> 
>           child: const Text("Delete"),
> 
>           onPressed: () {},
> 
>         ),
> 
>       ],
> 
>       child: ShadButton.ghost(
> 
>         icon: const Icon(
> 
>           LucideIcons.ellipsis,
> 
>           size: 16,
> 
>         ),
> 
>         onPressed: () {
> 
>           _isVisible.value = !_isVisible.value;
> 
>         },
> 
>       ),
> 
>     );
> 
>   }
> 
> 
> 
>   @override
> 
>   void dispose() {
> 
>     _isVisible.dispose();
> 
>     super.dispose();
> 
>   }
> 
> }
> 
> 
> 
> ```
> 
> 
> 
> However, when the context menu is dismissed by clicking outside, **_isVisible** will still be true, so I need to click on it twice to make it appear again.

Do you want a callback like `onVisibilityChange`?

**Comment by renanboni** [comment]
That would work, thanks!

**Comment by nank1ro** [comment]
@dickermoshe can you try visiting this page with your windows device?
https://shadcn-ui-playground.web.app/context-menu

Let me know if the issue is fixed.
I temporarely disabled the native context menu from anywhere, I'm waiting for a [response here](https://github.com/flutter/engine/pull/53278#issuecomment-2328309843) to disable it only for the context menu widget

**Comment by dickermoshe** [comment]
Fixed!


https://github.com/user-attachments/assets/d97733dd-39fd-4d2b-8b1f-0fdbda841a54

**Comment by nank1ro** [comment]
@renanboni just released the v0.9.1.
Use the new `controller` parameter to "open/close" the context menu.

**Comment by renanboni** [comment]
@nank1ro thanks a lot!
